src/core/api/internal.go

```diff
@@ -69,3 +76,96 @@ func (ia *InternalAPI) RenameAdmin() {
 	log.Debugf("The super user has been renamed to: %s", newName)
 	ia.DestroySession()
 }
+
+type QuotaSwitcher struct {
+	Disabled bool
+}
+
+// SwitchQuota ...
+func (ia *InternalAPI) SwitchQuota() {
+	var req QuotaSwitcher
+	if err := ia.DecodeJSONReq(&req); err != nil {
+		ia.SendBadRequestError(err)
+		return
+	}
+	// From disable to enable, update the quota usage bases on the DB records.
+	if config.QuotaPerProjectEnable() == false && req.Disabled == true {
+		if err := ensureQuota(); err != nil {
+			ia.SendBadRequestError(err)
+			return
+		}
+	}
+	defer config.GetCfgManager().Set(common.QuotaPerProjectEnable, req.Disabled)
+	return
+}
+
+func ensureQuota() error {
+	projects, err := dao.GetProjects(nil)
+	if err != nil {
+		return err
+	}
+	for _, project := range projects {
+		pSize, err := dao.CountSizeOfProject(project.ProjectID)
+		if err != nil {
+			logger.Warningf("error happen on counting size of project:%d , error:%v, just skip it.", project.ProjectID, err)
+			continue
+		}
+		afQuery := &models.ArtifactQuery{
+			PID: project.ProjectID,
+		}
+		afs, err := dao.ListArtifacts(afQuery)
+		if err != nil {
+			logger.Warningf("error happen on counting number of project:%d , error:%v, just skip it.", project.ProjectID, err)
+			continue
+		}
+		pCount := int64(len(afs))
+
+		// it needs to append the chart count
+		if config.GetCfgManager().Get(common.WithChartMuseum).GetBool() {
+			chartEndpoint := strings.TrimSpace(config.GetCfgManager().Get(common.ChartRepoURL).GetString())
+			if len(chartEndpoint) == 0 {
+				return errors.New("empty chartmuseum endpoint")
+			}
+			chartCtr, err := getChartCtr(chartEndpoint)
```

Comment:
There is a chart controller variable already in the package `api`: https://github.com/goharbor/harbor/blob/master/src/core/api/base.go#L147, you can use it directly